app/src/main/java/ca/unb/mobiledev/reflexrevolution/utils/InstructionUtil.java

```diff
@@ -19,16 +25,81 @@
         //Construct list of instructions based on gamemode
         switch(mode){
             case BASIC:
-                instructions.add(Instruction.BUTTON);
+                instructions.add(Instruction.TAP);
+                instructions.add(Instruction.DOUBLE_TAP);
+                instructions.add(Instruction.HOLD);
+                instructions.add(Instruction.DONT_TAP);
                 if (accelerometer != null) instructions.add(Instruction.SHAKE);
                 if (accelerometer != null && gravitySensor != null) instructions.add(Instruction.JUMP);
+
+                //Legacy instruction previously used for testing
+                //instructions.add(Instruction.BUTTON);
                 break;
 
             default:
                 break;
         }
         return instructions;
     }
+
+    //Display UI elements for the instruction passed in
+    public static void displayInstruction(Instruction instruction, ViewGroup layout, Context context){
+        //Label will be used in most cases, so initialize up here and set properties
+        //to make the rest cleaner
+        TextView label = new TextView(context);
+        label.setTextAlignment(View.TEXT_ALIGNMENT_CENTER);
+        label.setTextSize(TypedValue.COMPLEX_UNIT_SP, DEFAULT_TEXT_SIZE);
+        switch (instruction){
+            case TAP:
+                label.setText("TAP");
+                layout.addView(label);
+                break;
+
+            case DOUBLE_TAP:
+                label.setText("DOUBLE\nTAP");
+                layout.addView(label);
+                break;
+
+            case HOLD:
+                label.setText("HOLD");
+                layout.addView(label);
+                break;
```

Comment:
For the double-tap and hold instructions, it might be cool to use the same format as the don't tap. What I mean is that all of these instructions would have the main label "TAP" with a secondary label saying either "don't", "double", "hold" in a smaller font above.

That way the player really needs to pay attention since they will all look similar.

Reply:
Oh I like that idea! I'll throw it in.